libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts

```diff
@@ -51,77 +66,97 @@ export class RushPnpmCommandLineParser {
   private _debugEnabled: boolean;
   private _verboseEnabled: boolean;
 
-  public constructor(options: IRushPnpmCommandLineParserOptions) {
-    const { terminalProvider } = options;
-    this._debugEnabled = process.argv.indexOf('--debug') >= 0;
-    this._verboseEnabled = process.argv.indexOf('--verbose') >= 0;
-    const localTerminalProvider: ITerminalProvider =
-      terminalProvider ??
-      new ConsoleTerminalProvider({
-        debugEnabled: this._debugEnabled,
-        verboseEnabled: this._verboseEnabled
-      });
-    this._terminal = new Terminal(localTerminalProvider);
+  private constructor(
+    options: IRushPnpmCommandLineParserOptions,
+    terminal: ITerminal,
+    debugEnabled: boolean,
+    verboseEnabled: boolean
+  ) {
+    this._debugEnabled = debugEnabled;
+    this._verboseEnabled = verboseEnabled;
+    this._terminal = terminal;
+
+    // Are we in a Rush repo?
+    let rushConfiguration: RushConfiguration | undefined = undefined;
+    if (RushConfiguration.tryFindRushJsonLocation()) {
+      // showVerbose is false because the logging message may break JSON output
+      rushConfiguration = RushConfiguration.loadFromDefaultLocation({ showVerbose: false });
+    }
```

Comment:
Surprised we don't have a single call for this.

Reply:
Guess it isn't something that's really used all that often.